core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala

```diff
@@ -93,9 +93,8 @@ case class DeleteCommand(
       // Re-cache all cached plans(including this relation itself, if it's cached) that refer to
       // this data source relation.
       sparkSession.sharedState.cacheManager.recacheByPlan(sparkSession, target)
+      Seq(Row(this.metrics.get("numDeletedRows").map(_.value).getOrElse(0L)))
     }
-
-    Seq.empty[Row]
```

Comment:
let's see if this fixes your tests

Reply:
It doesn't, I dig some investigation and I found out that returning that Row makes the optimizer fails because it cannot guess the schema.

<img width="921" alt="Screen Shot 2022-06-30 at 10 12 56 AM" src="https://user-images.githubusercontent.com/812841/176777455-e10f68ef-50ad-4a28-976d-d39b271879dc.png">

Reply:
It doesn't, I dig some investigation and I found out that returning that Row makes the optimizer fails because it cannot guess the schema.

I took a screenshot (I tried to add a second field to the row to see what was the cause). This loop goes into an index array out of bound exception because the struct type has no fields and the converters is empty (see the indexes)

<img width="1436" alt="Screen Shot 2022-06-30 at 2 01 31 PM" src="https://user-images.githubusercontent.com/812841/176777806-66650b3a-2b99-4325-b5f2-50cddff30e95.png">
.

Reply:
We can fix this error by adding this (right before `innerChildren` will work) to define the output schema for `DeleteCommand`. 
```
 override val output: Seq[Attribute] = Seq(AttributeReference("numDeletedRows", LongType)())
```

Reply:
Thank you, this is very helpful. Is there some documentation around the relationship of the output attributes and the row returned by the command? Maybe we can introduce some sort of validation at the parent class to avoid errors in the catalyst optimizer

Reply:
I'm not sure if there's any documentation for this, but since this is just a spark class we're extending I don't think we should be introducing any validation (it would be outside of the delta codebase). We can check for the column `"numDeletedRows"` in the resulting df in the test suite.